### PR TITLE
Shim Conversion for xam_ui.cc

### PR DESCRIPTION
**Pull request by allens39**

### CODE REVIEW

**Comment by x1nixmzeng** [comment]
You've included another fix in the PR - was that intentional? If f2f0aa0 should be separate it can be reviewed properly

**Comment by Jeremy517** [comment]
He reverted that change, but he needs to squash the three commits into one so the history doesn't get polluted

File: src/xenia/kernel/xam/xam_ui.cc [hunk 1]
```diff
@@ -23,11 +23,11 @@ namespace xam {
 
 std::atomic<int> xam_dialogs_shown_ = {0};
 
-SHIM_CALL XamIsUIActive_shim(PPCContext* ppc_context,
-                             KernelState* kernel_state) {
+dword_result_t XamIsUIActive() {
   XELOGD("XamIsUIActive()");
```
Comment (DrChat) [inline src/xenia/kernel/xam/xam_ui.cc]: This logging line can die.

File: src/xenia/kernel/xam/xam_ui.cc [hunk 2]
```diff
@@ -318,19 +307,16 @@ dword_result_t XamShowDeviceSelectorUI(dword_t user_index, dword_t content_type,
 }
 DECLARE_XAM_EXPORT(XamShowDeviceSelectorUI, ExportTag::kImplemented);
 
-SHIM_CALL XamShowDirtyDiscErrorUI_shim(PPCContext* ppc_context,
-                                       KernelState* kernel_state) {
-  uint32_t user_index = SHIM_GET_ARG_32(0);
-
+dword_result_t XamShowDirtyDiscErrorUI(dword_t user_index) {
   XELOGD("XamShowDirtyDiscErrorUI(%d)", user_index);
```
Comment (DrChat) [inline src/xenia/kernel/xam/xam_ui.cc]: Ditto.

File: src/xenia/kernel/xam/xam_ui.cc [hunk 3]
```diff
@@ -121,16 +108,16 @@ SHIM_CALL XamShowMessageBoxUI_shim(PPCContext* ppc_context,
   XELOGD(
```
Comment (DrChat) [inline src/xenia/kernel/xam/xam_ui.cc]: Ditto.

**Pull request closed without merging**